python/lsst/ap/association/diaForcedSource.py

```diff
@@ -277,9 +277,7 @@ def _calibrate_and_merge(self,
         midPointTaiMJD = visit_info.getDate().get(system=DateTime.MJD)
         output_catalog["ccdVisitId"] = ccdVisitId
         output_catalog["midPointTai"] = midPointTaiMJD
-        # TODO DM-27170: fix this [0] workaround which gets a single character
-        # representation of the band.
-        output_catalog["filterName"] = diff_exp.getFilter().getCanonicalName()[0]
+        output_catalog["filterName"] = diff_exp.getFilterLabel().bandLabel
```

Comment:
Can you include a fallback for when `diff_exp.getFilterLabel() == None` (or equivalently `diff_exp.hasFilterLabel() == False`)? `""` seems like a reasonable value for that case (though that would also require special-casing physical-only filters). Same with the other assignments of `bandLabel`.

Reply:
Why shouldn't this just fail if there is no bandLabel? That seems like a recipe for subtly-buggy future catalogs.